MeshLib/MeshQuality/ElementQualityInterface.h

```diff
@@ -0,0 +1,105 @@
+/**
+ * \file   ElementQualityInterface.h
+ * \author Karsten Rink
+ * \date   2015-03-24
+ * \brief  Definition of the ElementQualityInterface class.
+ *
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef ELEMENTQUALITYINTERFACE_H_
+#define ELEMENTQUALITYINTERFACE_H_
+
+#include <vector>
+
+#include "BaseLib/Histogram.h"
+
+#include "MeshLib/Mesh.h"
+#include "MeshLib/MeshQuality/ElementQualityMetric.h"
+#include "MeshLib/MeshQuality/EdgeRatioMetric.h"
+#include "MeshLib/MeshQuality/ElementSizeMetric.h"
+#include "MeshLib/MeshQuality/SizeDifferenceMetric.h"
+#include "MeshLib/MeshQuality/AngleSkewMetric.h"
+#include "MeshLib/MeshQuality/RadiusEdgeRatioMetric.h"
+
+namespace MeshLib
+{
+
+/**
+ * Interface class for handling mesh element quality metrics
+ */
+class ElementQualityInterface
+{
+public:
+	/// Constructor
+	ElementQualityInterface(MeshLib::Mesh const& mesh, MeshQualityType t)
+	: _type(t), _mesh(mesh), _quality_tester(nullptr)
+	{
+		calculateElementQuality(_mesh, _type);
+	}
+
+	/// Destructor
+	~ElementQualityInterface() 
+	{ 
+		delete _quality_tester; 
+	}
+
+	/// Returns the vector containing a quality measure for each element.
+	std::vector<double> const& getQualityVector() const
+	{ 
+		return _quality_tester->getElementQuality(); 
+	}
+
+	/// Returns a histogram of the quality vector seperated into the given number of bins.
+	/// If no number of bins is specified, one will be calculated based on the Sturges criterium.
+	BaseLib::Histogram<double> getHistogram(std::size_t n_bins = 0) const
+	{
+		return _quality_tester->getHistogram(static_cast<size_t>(n_bins));
+	}
+
+	/// Writes a histogram of the quality vector to a specified file.
+	int writeHistogram(std::string const& file_name, std::size_t n_bins = 0) const
```

Comment:
I am not sure. I think I'd prefer to keep this method here to really have one interface that handles everything that is needed.